src/main/java/de/dennisguse/opentracks/ControllerFragment.java

```diff
@@ -0,0 +1,297 @@
+
+                ActivityUtils.vibrate(getContext(), 1000);
+            };
+            handlerUpdateTotalTime.postDelayed(buttonDelay, buttonDelayDuration);
+            return true;
+        }
+
+        //To stop a recording
+        if (viewBinding.controllerStop.equals(view)
+                && event.getAction() == MotionEvent.ACTION_DOWN
+                && recordingStatus.isRecording()) {
+
+
+            transition.startTransition(buttonDelayDuration);
+
+            ActivityUtils.vibrate(getContext(), 150);
+            viewBinding.controllerStatus.setText(R.string.hold_to_stop);
+
+            buttonDelay = () -> {
+                view.sendAccessibilityEvent(AccessibilityEvent.TYPE_VIEW_CLICKED);
+                getCallback().recordStop();
+                ActivityUtils.vibrate(getContext(), 1000);
+
+                transition.resetTransition();
+            };
+            handlerUpdateTotalTime.postDelayed(buttonDelay, buttonDelayDuration);
+            return true;
+        }
+
+        return false;
+    }
+
+    public void hide() {
+        viewBinding.controllerContainer.setVisibility(View.GONE);
+    }
+
+    public void show() {
+        viewBinding.controllerContainer.setVisibility(View.VISIBLE);
+    }
+
+    private void updateTotalTime() {
+        TrackRecordingService trackRecordingService = trackRecordingServiceConnection.getServiceIfBound();
+        if (trackRecordingService != null) {
+            totalTime = trackRecordingService.getTotalTime();
+        }
+    }
+
+    private void setTotalTime() {
+        if (totalTime != null) {
+            viewBinding.controllerTotalTime.setText(StringUtils.formatElapsedTimeWithHour(totalTime));
+        }
+    }
+
+    private void showStatusSetDefaultText() {
+        viewBinding.controllerStatus.setText(recordingStatus.isPaused() ? R.string.generic_paused : R.string.generic_recording);
+    }
+
+    private Callback getCallback() {
+        return (Callback) getActivity();
+    }
+
+    public interface Callback {
+        void recordStart();
+
+        void recordPause();
+
+        void recordStop();
+    }
+
+    private void onRecordingStatusChanged(TrackRecordingService.RecordingStatus status) {
+        recordingStatus = status;
+
+        if (!recordingStatus.isRecording()) {
+            handlerUpdateTotalTime.removeCallbacks(updateTotalTimeRunnable);
+        }
+        if (recordingStatus.isRecordingAndNotPaused()) {
+            handlerUpdateTotalTime.postDelayed(updateTotalTimeRunnable, UI_UPDATE_INTERVAL.toMillis());
+        }
+
+        viewBinding.controllerRecord.setImageResource(recordingStatus.isRecordingAndNotPaused() ? R.drawable.ic_button_pause : R.drawable.button_record);
+        viewBinding.controllerRecord.setContentDescription(getContext().getString(recordingStatus.isRecordingAndNotPaused() ? R.string.image_pause : R.string.image_record));
+
+        viewBinding.controllerStop.setEnabled(recordingStatus.isRecording());
```

Comment:
Stop button doesn't get disable until service is started. You have to start a recording and then stop it to see the stop button disabled on track list activity.

Reply:
Nice catch!
I forgot to initialize the UI - now it is happening!
Thanks!